### ISSUE TITLE
Additional options for fill tool

### ISSUE THREAD

**Issue by CandyFace** [pencil2d/pencil#762]
The fill tool needs more features to become really useful, one feature i'm particularly interested in implementing for bitmap is the ability to exclude fill on a specific color. 

The feature comes from the animation program Retas Studio, which in Asia was much loved when working with cels. 

![image](http://s1.postimg.org/blc8azjm7/tvp_ex.gif)

### TODO:
- [ ] Exclude/include colors when filling
- [ ] Close gap
- [ ] Expand/Contract fill

**Comment by Jose-Moreno** [comment]
@CandyFace Yes! There was a similar request done in the past, more like TVPaint where you could tint or delete specific colors via macros or a tool. But this feature distribution looks great too and is very close to the feature idea!

One thing though, this is considering that one would ink & paint with bitmap tools. But what could be done about the ink & paint with vector tools?. I just realized it would be interesting to propose a feature where you could tint specific segments between control points, to achieve a similar effect.
It would be similar to flash or toonboom where you could do just that, select a stroke segment and change it's colour. But if we could adapt this Retas idea to that, it's be awesome to maintain a single interface for both solutions. A layer-aware option panel for the fill tool.

**Comment by CandyFace** [comment]
I have no real plans for the vector fill tool except fixing the fill algorithm by filling the given vector points ( which I already have sorta working ) and stroke fill. Filling segments could be an interesting idea too.

You can see WIP here
![2](https://user-images.githubusercontent.com/1045397/30774327-1688feca-a081-11e7-81d4-39000ddffde4.gif)

It's not perfect yet, because it works like this:
1. Draw some segment
2. intersect the mouse with the contour, this extracts the vector points
3. press anywhere to fill the shape. 

It's fast and accurate because it simply follows the points, however it doesn't stop filling when another color intersects, compared to pixel fill.

**Comment by CandyFace** [comment]
I have reliable vector fill working now. 
objects sticking to each other has also been fixed in most cases.
![vector progress](https://user-images.githubusercontent.com/1045397/30879959-553c5144-a301-11e7-97e5-d1468e91adb5.gif)

I still need to solve objects sticking each other with automatic contour fill but it's getting there.
